### PR TITLE
Add contributing guidelines

### PR DESCRIPTION
**Pull request by SamDuffield**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -83,3 +83,20 @@ In particular, the
...
+
+Pull requests are welcomed!
+1. First fork the repo and create your branch from [`develop`](https://github.com/CQCL/qujax/tree/develop).
+2. Add your code.
+3. Add your tests.
+4. Update the documentation if required.
+5. Issue a pull request into [`develop`](https://github.com/CQCL/qujax/tree/develop).
+
+New commits on [`develop`](https://github.com/CQCL/qujax/tree/develop) will then be merged into
+[`main`](https://github.com/CQCL/qujax/tree/develop) on the next release.
```
Comment (cqc-melf) [inline README.md:98]: The link that should be to main is to develop

File: README.md [hunk 2]
```diff
@@ -83,3 +83,20 @@ In particular, the
...
+
+We use [semantic versioning](https://semver.org/) with version number `x.y.z` -
```
Comment (cqc-melf) [inline README.md:100]: Is this what you have planed to do? I would recommend staying with 0.x.y until you have established a stable api?

File: README.md (hunk 1 above)
Comment (SamDuffield) [reply README.md:98]: good spot!

File: README.md (hunk 2 above)
Comment (SamDuffield) [reply README.md]: removed comment on versioning

**Review by cqc-melf — APPROVED** [review]

**Pull request merged**